src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/ValidationAttributeStore.cs

```diff
@@ -230,8 +232,61 @@ internal bool TryGetPropertyStoreItem(string propertyName, [NotNullWhen(true)] o
                     propertyStoreItems[property.Name] = item;
                 }
 
+                return CombineMetadataTypeAttributes(propertyStoreItems);
+            }
+
+            [return: DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)]
+            private Dictionary<string, PropertyStoreItem> CombineMetadataTypeAttributes(Dictionary<string, PropertyStoreItem> propertyStoreItems)
+            {
+                // check if metadata attribute is present on the type
+                MetadataTypeAttribute? attribute = ((MetadataTypeAttribute[])Attribute.GetCustomAttributes(_type, typeof(MetadataTypeAttribute))).FirstOrDefault();
+                if (attribute != null)
+                {
+                    // get properties of metadata type
+                    TypeInfo t = (TypeInfo)attribute.MetadataClassType;
+                    var properties = t.GetProperties()
+                        .Where(prop => !prop.GetIndexParameters().Any());
+
+                    foreach (PropertyInfo property in properties)
+                    {
```

Comment:
```suggestion
                    foreach (PropertyInfo property in t.GetProperties())
                    {
                        if (property.GetIndexParameters().Length == 0)
                        {
```

Reply:
Should you also be filtering `static` properties? `GetProperties()` returns both instance and static properties.

Reply:
I've incorporated your suggested code. Is the reasoning of the style change to reduce memory and object iteration for performance reasons?

Reply:
@eerhardt Re `static` properties: Only public static properties will be processed. This is the same behavior as in .Net Framework. My branch finally got through all the pipelines, yay, progress. Thanks for all your help.

Reply:
>  Is the reasoning of the style change to reduce memory and object iteration for performance reasons?

Yes. In general, we have been moving away from using Linq in the BCL. See #44923 #44825 #47496 #47873 #44734

Reply:
I would hope the area owners would review this change. I'm just here reviewing because I was pinged on the issue, and was helping out on the "trim compatibility" parts of it.

@ajcvickers

Reply:
Thanks. Would it be appropriate for you to assign the review to ajcvickers?